crates/configuration/src/network.rs

```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
         // This regex is used to replace the "" enclosed u128 value to a raw u128 because u128 is not supported for TOML serialization/deserialization.
-        let re = Regex::new(r#""U128%(?<u128_value>\d+)""#).expect("regex should be valid, this is a bug please report it: https://github.com/paritytech/zombienet-sdk/issues");
+        let re = Regex::new(r#""U128%(?<u128_value>\d+)""#)
+            .expect(&format!("{} {}", VALID_REGEX, THIS_IS_A_BUG));
         let toml_string = toml::to_string_pretty(&self)?;
 
         Ok(re.replace_all(&toml_string, "$u128_value").to_string())
     }
+
+    pub fn load_from_toml(path: &str) -> Result<NetworkConfig, anyhow::Error> {
+        let file_str = fs::read_to_string(path).expect(&format!("{} {}", RW_FAILED, THIS_IS_A_BUG));
+        let re: Regex = Regex::new(r"(?<field_name>(initial_)?balance)\s+=\s+(?<u128_value>\d+)")
+            .expect(&format!("{} {}", VALID_REGEX, THIS_IS_A_BUG));
+
+        let mut network_config: NetworkConfig = toml::from_str(
+            re.replace_all(&file_str, "$field_name = \"$u128_value\"")
+                .as_ref(),
+        )?;
+
+        if network_config.relaychain.is_none() {
+            Err(anyhow!("Relay chain does not exist."))?
+        }
+
+        // retrieve the defaults relaychain for assigning to nodes if needed
+        let relaychain_default_command: Option<Command> = network_config
+            .relaychain
+            .as_ref()
+            .unwrap()
+            .default_command()
+            .cloned();
+
+        let relaychain_default_image: Option<Image> = network_config
+            .relaychain
+            .as_ref()
+            .unwrap()
+            .default_image()
+            .cloned();
+
+        let relaychain_default_db_snapshot: Option<crate::shared::types::AssetLocation> =
+            network_config
+                .relaychain
+                .as_ref()
+                .unwrap()
+                .default_db_snapshot()
+                .cloned();
+
+        let default_args: Vec<Arg> = network_config
+            .relaychain()
+            .default_args()
+            .into_iter()
+            .cloned()
+            .collect();
+
+        let mut nodes: Vec<NodeConfig> = network_config
+            .relaychain()
+            .nodes()
+            .into_iter()
+            .cloned()
+            .collect();
+
+        // Validation checks for relay
+        TryInto::<Chain>::try_into(network_config.relaychain().chain().as_str())?;
+        if relaychain_default_image.is_some() {
+            TryInto::<Image>::try_into(relaychain_default_image.clone().expect(VALIDATION_CHECK))?;
+        }
+        if relaychain_default_command.is_some() {
+            TryInto::<Command>::try_into(
+                relaychain_default_command.clone().expect(VALIDATION_CHECK),
+            )?;
+        }
+
+        // SAFETY: is ok to use `unwrap` here since we ensure that is some at the begging of this fn
```

Comment:
```suggestion
        // SAFETY: is ok to use `unwrap` here since we ensure that is some at the begging of this fn
```
I think this comment is outdated (needs to be a few lines before :))